clang-tools-extra/clang-tidy/utils/ASTUtils.cpp

```diff
@@ -96,7 +96,7 @@ bool areStatementsIdentical(const Stmt *FirstStmt, const Stmt *SecondStmt,
   if (FirstStmt == SecondStmt)
     return true;
 
-  if (FirstStmt->getStmtClass() != FirstStmt->getStmtClass())
+  if (FirstStmt->getStmtClass() != SecondStmt->getStmtClass())
```

Comment:
This looks like a functional change, it would be good to have a test for it.

Reply:
@PiotrZSL If I removed this condition and ninja target still passes. Shall I remove this or can you provide a test case or guidance? I have this function is used in BranchCloneCheck.cpp but I am not sure how the test case look like.